rules/checks.js

```diff
@@ -0,0 +1,187 @@
+const { parse } = require('@connectedcars/react-i18n/dist/parser')
+
+const defaultOptions = {
+  globalData: [],
+  replaceStringRegex: {
+    pattern: '{__KEY__}',
+  },
+  expressions: {
+    t: ['singular', 'data', 'context'],
+    tn: ['count', 'singular', 'plural', 'data', 'context'],
+    tx: ['singular', 'data', 'context'],
+    tnx: ['count', 'singular', 'plural', 'data', 'context'],
+  },
+}
+
+module.exports = {
+  create(context) {
+    const options = {
+      ...defaultOptions,
+      ...context.options[0],
+    }
+
+    const strictPattern = '[\\w-]+'
+
+    return {
+      CallExpression(node) {
+        const funcName = node.callee.name
+
+        const args = options.expressions[funcName]
+        if (!args) {
+          return
+        }
+
+        const singularIndex = args.indexOf('singular')
+        const singular = node.arguments[singularIndex]
+        if (!singular) {
+          return context.report({
+            node,
+            message: 'Singular function defined but no singular string found',
```

Comment:
for the user it would perhaps to be better to explictly state it's the translation function. It can be a bit difficult to see the eslint-rule. Same below
```suggestion
            message: 'Singular translation function defined but no singular string found',
```